### PR TITLE
2 fa works

### PR DESCRIPTION
**Pull request by karensawrey**
2FA text, iteration 1.
Next pre-planned steps: 
* Add screenshots from an authenticator app
* Add info on reconfiguring 2FA

### CODE REVIEW

File: pages/pipelines/secrets.md.erb [hunk 1]
```diff
@@ -155,3 +155,49 @@ steps:
...
+<%= image "2fa-3.png", width: 380, height: 238, alt: "Screenshot of Password Confirmation" %>
+<br>
+Enter your Buildkite account password and proceed.
+<%= image "2fa-2.png", width: 850, height: 372, alt: "Screenshot of Accessing Two-Factor Authentication" %>
+
+Click the Setup Two-Factor Authentication button to start securing your Buildkite account. 
+<%= image "2fa-4.png", width: 1172, height: 480, alt: "Screenshot of Setup Button for Two-Factor Authentication" %>
+
+**Step 1: Store recovery codes.**  
+You will need them in case you lose access to your authenticator application to restore access to your account. Use the buttons to either copy the codes to Clipboard or download them as a text file. Keep your recovery codes in a safe digital space or print them out and hide them well. Never share your recovery codes.
```
Comment (plaindocs) [inline pages/pipelines/secrets.md.erb:177]:
```suggestion
You will need them to restore access to your account if you lose access to your authenticator application. Use the buttons to either copy the codes to Clipboard or download them as a text file. Keep your recovery codes in a safe digital space or print them out and hide them well. Never share your recovery codes. 
```

this reads a little better to me

File: pages/pipelines/secrets.md.erb [hunk 2]
```diff
@@ -155,3 +155,49 @@ steps:
...
+**Step 2: Configure Authenticator Application**  
+To activate two-factor authentication, scan the barcode that appears in the Buildkite dashboard with the authenticator application of your choice. If you cannot scan the barcode, you can use the secret key below the barcode.
+
+After you've scanned the barcode or activated the authenticator application using the secret key, Buildkite will appear on the list of accounts registered in that application. Your authenticator will provide a new randomly generated six-digit code (your One Time Pass) roughly every 30 seconds. Enter this code into the corresponding field in the Buildkite app and click Activate. 
+
+<%= image "2fa-6.png", width: 604, height: 728, alt: "Screenshot of Barcode and Secret Key for Two-Factor Authenticator" %>
+
+Congratulations! You have now successfully enabled the two-factor authentication for your Buildkite account. This will be confirmed by an 'Enabled' badge next to the Two-Factor Authentication option in your Personal Settings. 
+
+<%= image "2fa-7.png", width: 1172, height: 350, alt: "Screenshot of Two-Factor Authentication Enabled Badge" %>
```
Comment (plaindocs) [inline pages/pipelines/secrets.md.erb:192]: Did you obscure the barcode as well as key? :-p

File: pages/pipelines/secrets.md.erb [hunk 3]
```diff
@@ -155,3 +155,49 @@ steps:
...
+
+You can set up two-factor authentication in the Buildkite dashboard. To do it, click your profile picture and select Personal Settings in the drop-down menu. 
+
+<%= image "2fa-1.png", width: 710, height: 262, alt: "Screenshot of Personal Settings Button" %>
+
+Next, navigate to the Two-Factor Authentication tab and click it (you may be asked to enter your password in the Confirm Password field).
+<%= image "2fa-3.png", width: 380, height: 238, alt: "Screenshot of Password Confirmation" %>
+<br>
+Enter your Buildkite account password and proceed.
+<%= image "2fa-2.png", width: 850, height: 372, alt: "Screenshot of Accessing Two-Factor Authentication" %>
```
Comment (plaindocs) [inline pages/pipelines/secrets.md.erb:171]: Might wanna obscure the email address here?

File: pages/pipelines/secrets.md.erb [hunk 4]
```diff
@@ -155,3 +155,49 @@ steps:
...
+
+Click the Setup Two-Factor Authentication button to start securing your Buildkite account.
```
Comment (plaindocs) [inline pages/pipelines/secrets.md.erb:173]: PR to fix the button name here https://github.com/buildkite/buildkite/pull/5987

**Review by plaindocs — APPROVED** [review]
This is great! I've left a few minor comments inline, and the more I think I about it the more I think *Tutorials* is a better temporary home for it than the security page.

File: pages/pipelines/secrets.md.erb [hunk 5]
```diff
@@ -155,3 +155,49 @@ steps:
...
+<%= image "2fa-7.png", width: 1172, height: 350, alt: "Screenshot of Two-Factor Authentication Enabled Badge" %>
+
+Next time you try to log into your Buildkite account from a new browser, device, or location, you will be asked to enter the current One Time Password provided by your authentication app.
+
+You can always reconfigure or deactivate the 2FA if you need to. This can be done in the Two-Factor Authentication tab in Personal Settings for your Buildkite account in the dashboard.  
+
+[1Password]: <https://support.1password.com/one-time-passwords/>
+[OTP Auth]: <https://cooperrs.de/otpauth.html>
+[Authy]: <https://authy.com/>
+[Duo Mobile]: <duo.com/product/multi-factor-authentication-mfa/two-factor-authentication-2fa>
```
Comment (plaindocs) [inline pages/pipelines/secrets.md.erb:201]: And this needs an `https` and is what is causing the Rspec link error.

(the spell checker is also confused by `Authy`, which needs adding to `vale/vocab.txt`

**Comment by plaindocs** [comment]
Couple of extra things to think about:

- [ ] Can you enforce 2FA org wide

image

- [ ] What to do when you're locked out and have no codes.

File: pages/pipelines/secrets.md.erb [hunk 6]
```diff
@@ -155,3 +155,49 @@ steps:
         https://api.github.com/repos/my-org/my-app/deployments
 ```
 {: codeblock-file="pipeline.yml"}
+
+## Two-Factor Authentication
```
[the remark this answers is not in the document]
Comment (karensawrey) [reply pages/pipelines/secrets.md.erb:159]: Thank you @plaindocs! Highly appreciated.
I'm all for creating a separate page in Tutorials for it, but I think that in this case, it's also worth linking to it from the Secrets page. On it now, alongside all the other suggestions and fixes that you've introduced.

File: pages/pipelines/secrets.md.erb [hunk 7]
```diff
@@ -155,3 +155,49 @@ steps:
...
+
+Two-factor authentication (2FA) can be added to provide an additional layer of security to your Buildkite account and make sure you're the only person who has access to handling your builds. Once 2FA is enabled on your Buildkite account, the only way to login to your account is by knowing both your password and a unique code generated by a third-party application such as [1Password], [OTP Auth], [Duo Mobile], [Authy], or [Google Authenticator].
```
[the remark this answers is not in the document]
Comment (karensawrey) [reply pages/pipelines/secrets.md.erb:161]: Ah, I get the idea & I'll reword it. What I wanted to put across is that if someone gets a hold of the password, it won't do them any good without getting access to the OTP.

File: pages/pipelines/secrets.md.erb (hunk 1 above)
Comment (karensawrey) [reply pages/pipelines/secrets.md.erb:177]: Amen! Thanks.

File: pages/pipelines/secrets.md.erb (hunk 2 above)
Comment (karensawrey) [reply pages/pipelines/secrets.md.erb:192]: Nah, but I had some thoughts about changing this screenshot altogether (or adding an extra one without the OTP in the field so I thought I'd deal with it later :)

File: pages/pipelines/secrets.md.erb (hunk 3 above)
Comment (karensawrey) [reply pages/pipelines/secrets.md.erb:171]: Yep, you're right, thanks!

File: pages/pipelines/secrets.md.erb (hunk 5 above)
Comment (karensawrey) [reply pages/pipelines/secrets.md.erb:201]: Will deal with duo.com & Authy.

**Comment by plaindocs** [comment]
What to do when locked out

> I will need to get verification from an admin that they agree with us removing the 2FA, then I can set up video call where you will need to present some government-issued photo identification so that I can verify your identity. Alternatively, you could ask an admin to remove your user account from your organisation in Buildkite and I can then delete it which will allow you to create a new one.

@jayco do you have a preference for which of these should be the main option? Second probs?

**Comment by jayco** [comment]
> What to do when locked out
> 
> > I will need to get verification from an admin that they agree with us removing the 2FA, then I can set up video call where you will need to present some government-issued photo identification so that I can verify your identity. Alternatively, you could ask an admin to remove your user account from your organisation in Buildkite and I can then delete it which will allow you to create a new one.
> 
> @jayco do you have a preference for which of these should be the main option? Second probs?

Yup - definitely the second 👌

**Comment by plaindocs** [comment]
You can add `Authy` to the `vale/vocab.txt`in this PR, so that it builds without an error.

File: pages/tutorials/2fa.md.erb [hunk 8]
```diff
@@ -0,0 +1,67 @@
+# Two-Factor Authentication  
+
+Two-factor authentication (2FA) can be added to your Buildkite account to provide an additional layer of security and to make sure your builds are safe even if your login credentials are compromised (exposed or stolen). Once 2FA is enabled on your Buildkite account, the only way to log in to your account is by knowing both your password and a unique code generated by a third-party application such as [1Password], [OTP Auth], [Duo Mobile], [Authy], or [Google Authenticator].  
+
+{:toc}
```
Comment (plaindocs) [inline pages/tutorials/2fa.md.erb:5]:
```suggestion
{:toc}
```

**Comment by plaindocs** [comment]
@karensawrey I think we can merge this in right?

**Review by plaindocs — APPROVED** [review]

**Pull request merged**